File: week4/Java/1290. Convert Binary Number in a Linked List to Integer.java
class Solution {
    public int getDecimalValue(ListNode head) {
        int num = head.val;
        while (head.next != null) {
            num = num * 2 + head.next.val;  // need to remember this logic.....
            head = head.next;    
        }
        return num;
    }
}

File: week4/python/1290. Convert Binary Number in a Linked List to Integer.py
class Solution:
    def getDecimalValue(self, head: ListNode) -> int:
        result = 0
        while head: # linked list lo manam end daaka traverse kaavali soo ee condition
            result = result * 2 + head.val # idhi amo manam result anae variable loki mana binary value ni integer laaga store chaestunnam (iteratively)
            head = head.next  # increment chaestunam
        return result  # returning result
